workflows/qe/satellite6-foreman-maintain-automation.groovy

```
@@ -0,0 +1,172 @@
+                    def command = 'pytest -v --junit-xml=foreman-results.xml --ansible-host-pattern "${COMPONENT}" --ansible-user root --ansible-inventory testfm/inventory '
+                    try {
+                        if ("${PYTEST_OPTIONS}") {
+                            sh_venv command + '${PYTEST_OPTIONS}'
+                        }
+                        else if ("${PYTEST_MARKS}") {
+                            sh_venv command + 'tests/ -m ${PYTEST_MARKS}'
+                        }
+                        else {
+                            sh_venv command + 'tests/'
+                        }
+                    }
+                    catch(all) {
+                        currentBuild.result = 'UNSTABLE'
+                    }
+                }
+            }
+        }
+
+        stage("Post Testing - Check Errors in Logs") {
+            steps {
+                script {
+                    withCredentials([sshUserPrivateKey(credentialsId: 'id_hudson_rsa', keyFileVariable: 'identity', passphraseVariable: '', usernameVariable: 'userName')]) {
+                        remote = [: ]
+                        remote.name = "Satellite Server "
+                        remote.allowAnyHosts = true
+                        remote.host = SERVER_HOSTNAME
+                        remote.user = userName
+                        remote.identityFile = identity
+
+                        if ("${TEST_UPSTREAM}" != "true") {
+                            sshCommand remote: remote, command: 'cat /var/log/foreman-maintain/foreman-maintain.log | grep -i error'
+                        }
+                        else {
+                            sshCommand remote: remote, command: 'cat ~/foreman_maintain/logs/foreman-maintain.log | grep -i error'
+                        }
+                    }
+                }
+            }
+        }
+
+        stage("Generating Report") {
+            steps {
+                sh_venv '''
+                set +e
```

Why are we suppressing error here?. 
If this fails, job should fail.

If we want this behaviour , add this to post directive